### PR TITLE
refactor rate model to support curve model

### PR DESCRIPTION
**Pull request by yz89**
- remove APR struct
- modify the `InterestRateModel` type from struct to enum
- support curve rate model

### CODE REVIEW

File: pallets/loans/src/lib.rs [hunk 1]
```diff
@@ -356,13 +356,18 @@ pub mod pallet {
...
                     self.base_rate,
-                    self.kink_rate,
+                    self.jump_rate,
                     self.full_rate,
-                    self.kink_utilization,
+                    self.jump_utilization,
                 );
-                interest_model.check_parameters().unwrap();
+                if !interest_model.check_model() {
+                    log::info!(
```
Comment (ayushmishra2005) [inline pallets/loans/src/lib.rs:366]: Shouldn't this be log.error?

File: node/parallel/src/chain_spec/heiko.rs [hunk 2]
```diff
@@ -185,9 +185,9 @@ fn testnet_genesis(
             borrow_index: Rate::one(),                             // 1
             exchange_rate: Rate::saturating_from_rational(2, 100), // 0.02
             base_rate: Rate::saturating_from_rational(2, 100),     // 2%
-            kink_rate: Rate::saturating_from_rational(10, 100),    // 10%
+            jump_rate: Rate::saturating_from_rational(10, 100),    // 10%
```
Comment (kaichaosun) [inline node/parallel/src/chain_spec/heiko.rs:188]: These model field can be grouped use a struct or tuple, can be reactored later.

**Review by kaichaosun — APPROVED** [review]
👍

**Review by c410-f3r — APPROVED** [review]
`pallets/loans/src/rate.rs` was transposed to the `InterestRateModel::Jump` enum variant so it should only be a matter of implement the remaining functions for `InterestRateModel::CurveModel`.

Thanks for doing this introductory work for #227

**Comment by yz89** [comment]
> `pallets/loans/src/rate.rs` was transposed to the `InterestRateModel::Jump` enum variant so it should only be a matter of implement the remaining functions for `InterestRateModel::CurveModel`.
> 
> Thanks for doing this introductory work for #227

Yes, you are right!

**Review by GopherJ — APPROVED** [review]

File: node/parallel/src/chain_spec/heiko.rs (hunk 2 above)
Comment (yz89) [reply node/parallel/src/chain_spec/heiko.rs:188]: Sure

File: pallets/loans/src/lib.rs (hunk 1 above)
Comment (yz89) [reply pallets/loans/src/lib.rs]: Thanks, I've replaced it with panic.

**Pull request merged**